snippets/vectorAngle.md

```diff
@@ -0,0 +1,28 @@
+---
+title: vectorAngle
+tags: math, beginner
+---
+
+Returns the angle (theta) value between two vectors.
+
+Use `Math.trunc()` to convert the half of `length` to integer. 
+Use `Array.prototype.slice()` slice the incoming array into two vectors. 
+Use `Array.prototype.reduce()`, `Math.pow()` and `Math.sqrt()` to calculate the magnitude of each vector and the scalar product of the two vectors.
+Use `Math.acos()` to calculate arccos and get the theta.
+
+
+```js
+const vectorAngle = (...x) =>{ 
+  let half = Math.trunc(x.length / 2);
+  let [a, b] = [x.slice(0, half), x.slice(half)];  
```

Comment:
Assuming you remove the above calculation and early return, using `length/2` here should be fine and will shorten the code a little.